packages/host/app/components/operator-mode/profile/profile-settings-modal.gts

```diff
@@ -250,6 +389,42 @@ export default class ProfileSettingsModal extends Component<Signature> {
     }
   });
 
+  private async onSavePassword() {
+    if (!this.currentPassword) {
+      throw new Error(
+        'bug: should never get here: current password is required',
+      );
+    } else if (!this.newPassword) {
+      throw new Error('bug: should never get here: new password is required');
+    }
+
+    try {
+      let auth = {
+        type: 'm.login.password',
+        user: this.matrixService.userId,
+        password: this.currentPassword,
+        identifier: {
+          type: 'm.id.user',
+          user: this.matrixService.userId,
+        },
+      } as IAuthData & { type: string };
+      await this.matrixService.client.setPassword(auth, this.newPassword);
+      this.resetPasswordFields();
+      this.submode = undefined;
+    } catch (e) {
+      this.currentPasswordError = 'Current password is invalid';
```

Comment:
There are other ways this can fail, not only when the current password is invalid. Add some separation here, i.e. a general unknown error and invalid password error (known error)